ganga/GangaGUI/gui/routes.py

```diff
@@ -64,13 +83,365 @@ def decorated(*args, **kwargs):
+
+    # Function for action: resubmit
+    if action == "resubmit":
+
+        # resubmit the job
+        try:
+            j = jobs(job_id)
+            j.resubmit()
+        except Exception as err:
+            return jsonify({"success": False, "message": str(err)}), 400
+
+    # Function for action: runPostProcessors
+    if action == "runPostProcessors":
+
+        # runPostProcessors of the job
+        try:
+            j = jobs(job_id)
+            j.runPostProcessors()
+        except Exception as err:
+            return jsonify({"success": False, "message": str(err)}), 400
+
+    # Function for action: submit
+    if action == "submit":
+
+        # submit the job
+        try:
+            j = jobs(job_id)
+            j.submit()
+        except Exception as err:
+            return jsonify({"success": False, "message": str(err)}), 400
+
+    return jsonify(
+        {"success": True, "message": "Successfully completed the action {} on the Job ID {}".format(action, job_id)})
+
+
+# Delete Job API - DELETE Method
+@app.route("/job/<int:job_id>", methods=["DELETE"])
 @token_required
-def resource(current_user):
-    return jsonify({"success": True, "message": "Token is valid"})
+def job_delete_endpoint(current_user, job_id: int):
+    """
+    Given the job id, removes the job from the job repository.
 
+    :param job_id: int
+    """
+
+    # Imports
+    from GangaCore.GPI import jobs
+
+    # Job existence check
+    if job_id not in list(jobs.ids()):
+        return jsonify({"success": False, "message": "Job with ID {} does not exist".format(job_id)}), 400
+
+    # Remove job
+    try:
+        j = jobs[job_id]
+        j.remove()
+    except Exception as err:
+        return jsonify({"success": False, "message": str(err)}), 400
+
+    return jsonify({"success": True, "message": "Job with ID {} successfully removed".format(job_id)})
+
+
+# ******************** Helper Functions ******************** #
+
+def get_job_data(job_id: int) -> dict:
+    """
+    Given the job_id, return a dict containing
+    [id, fqid, status, name, subjobs, application, backend, backend.actualCE, comments, subjob_statuses] as dict keys and their values.
+
+    :param job_id: int
+    :return: dict
+    """
+    from GangaCore.GPI import jobs
+    # Get job from the job list
+    try:
+        j = jobs[int(job_id)]
+    except Exception as err:
+        return jsonify({"success": False, "message": str(err)})
+
+    # Store job info in a dict
+    job_data = {}
+    job_data["id"] = str(j.id)
+    job_data["fqid"] = str(j.fqid)
+    job_data["status"] = str(j.status)
+    job_data["name"] = str(j.name)
+    job_data["subjobs"] = len(j.subjobs)
+    job_data["application"] = str(type(j.application))
+    job_data["backend"] = str(type(j.backend))
+    job_data["backend.actualCE"] = str(j.backend.actualCE)
+    job_data["comment"] = str(j.comment)
+    job_data["subjob_statuses"] = str(j.returnSubjobStatuses())
```

Comment:
Maybe again some way to get this information automatically.

Reply:
This is just creating the response object to be returned. As each the values to be sent back as the response in JSON format needs to be `str`. It's just making sure that the values are `str` and the server doesn't give an error.

Reply:
Since you already know the information you are going to show: `id, fqid, status, name, subjobs, application, backend, backend.actualCE, comments, subjob_statuses]`, I believe a good strat would be to loop over the displayed attr and use `getattr` to fetch them..
```python
for attr in [id, fqid, status, name, subjobs, application, backend, backend.actualCE, comments, subjob_statuses]:
  value = getattr(job, attr)
  job_data[attr] = value
```

Reply:
Same can be done with the multiple `functions` called in `/action`. 
```rust
getattr(job, actioname)() # equivalent to job.actioname()
```

Reply:
This is really helpful, thanks a lot @DumbMachine